apps/emqx/src/emqx_logger_textfmt.erl

```diff
@@ -18,22 +18,77 @@
 
 -export([format/2]).
 -export([check_config/1]).
+-export([try_format_unicode/1]).
 
 check_config(X) -> logger_formatter:check_config(X).
 
-format(#{msg := {report, Report}, meta := Meta} = Event, Config) when is_map(Report) ->
-    logger_formatter:format(Event#{msg := {report, enrich(Report, Meta)}}, Config);
-format(#{msg := Msg, meta := Meta} = Event, Config) ->
-    NewMsg = enrich_fmt(Msg, Meta),
-    logger_formatter:format(Event#{msg := NewMsg}, Config).
+format(#{msg := {report, Report0}, meta := Meta} = Event, Config) when is_map(Report0) ->
+    Report1 = enrich_report_mfa(Report0, Meta),
+    Report2 = enrich_report_clientid(Report1, Meta),
+    Report3 = enrich_report_peername(Report2, Meta),
```

Comment:
clientid and peername may set in process dictionary (as logger metadata), we need to merge them ?

Reply:
Before: 
we define emqx_logger_textfmt's template in emqx_conf_shcema, Use placeholders(clientid,peername) to format messages, but the logger_formatter format placeholder as '~p', we need format by ~'ts":
like this:
```erlang
[time," [",level,"] ",
                {clientid,
                    [{peername,
                        [clientid,"@",peername," "],
                        [clientid, " "]}],
                    [{peername,
                        [peername," "],
                        []}]},
                msg,"\n"],
```
Now we remove clientid and peername from template.
All the log format is according to the template, if the template does not use the placeholder in the meta, It can't format by logger_formatter.

```erlang
[time," [",level,"] ", msg,"\n"],
```
So, we need enrich the clientid, peername, topic by the right format in to the msg(msg placeholder is in the template).

Reply:
I know.
But not all SLOG macros provide all metadata fields.
Because the old formatter had logger template fill it in.
But now since we don’t use the template any more, well need to activate read the meta in the formatter code